src/main/java/io/stargate/sgv2/jsonapi/service/operation/model/impl/CreateCollectionOperation.java

```diff
@@ -182,7 +186,33 @@ private Uni<Supplier<CommandResult>> executeCollectionCreation(QueryExecutor que
               } else {
                 return new SchemaChangeResult(true);
               }
-            });
+            })
+        .onFailure(
+            error ->
+                error instanceof InvalidQueryException
+                    && error
+                        .getMessage()
+                        .matches(
+                            ".*Cannot have more than \\d+ indexes, failed to create index on table.*"))
```

Comment:
This may be necessary, but I fear that error message may vary across different versions of backend; we saw this a lot with Stargate REST where C-3.11/4.0/DSE had sometimes small, sometimes big differences in fail messages.
If there was error code of some kind it'd better to check that.

Reply:
Looking at the logic very concerned with dropping table for invalid query. Should we do read on table with limit 1 and delete collection only on a successful empty result set. Dropping collection because of a failed query need more proof checking. @amorton @jeffreyscarpenter Thoughts?

Reply:
@maheshrajamani the intent is that we only do the rollback in the case of a too many indexes error, which it appears this is attempting to do. To @tatu 's point, do we not have an error code yet at this point in the flow that we can check?

Reply:
My concern was around what if some user tries to recreate existing collection and index creation statement fails and we end up dropping the table.

Reply:
table dropping will only occurs when DB guardrail is violated.
`If a user recreates an existing collection`, I assume you are saying this existing collection do not have all the sai indexes? In this case, the collection is invalid in our logic, if ending up dropping the table, it would be reasonable, IMO.

Reply:
Regarding the error code, unfortunately no.
The only thing driver has is error msg.

```

Exception in thread "main" com.datastax.oss.driver.api.core.servererrors.InvalidQueryException: Cannot have more than 50 indexes, failed to create index on table with CREATE CUSTOM INDEX IF NOT EXISTS text5_index1 ON collection_test1 (text5) USING 'StorageAttachedIndex'
        at com.datastax.oss.driver.api.core.servererrors.InvalidQueryException.copy(InvalidQueryException.java:48)
        at com.datastax.oss.driver.internal.core.util.concurrent.CompletableFutures.getUninterruptibly(CompletableFutures.java:149)
        at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:53)
        at com.datastax.oss.driver.internal.core.cql.CqlRequestSyncProcessor.process(CqlRequestSyncProcessor.java:30)
        at com.datastax.oss.driver.internal.core.session.DefaultSession.execute(DefaultSession.java:230)
        at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:54)
        at com.datastax.oss.driver.api.core.cql.SyncCqlSession.execute(SyncCqlSession.java:78)
        at org.example.VectorTest.main(VectorTest.java:64)
```